File: snippets/syntax/SyntaxTest.go
// Sample Go file for testing VSCode theme syntax highlighting
package main

import (
	"fmt"
	"sync"
	"time"
)

// Constants
const (
	MaxCount = 100
	Greeting = "Hello, Go!"
)

// Struct definition
type SyntaxTest struct {
	name  string
	count int
	items []string
}

// Interface
type Printable interface {
	PrintInfo()
}

// Struct method
func (s *SyntaxTest) PrintInfo() {
	fmt.Printf("Name: %s, Count: %d, Items: %v\n", s.name, s.count, s.items)
}

// Function with variadic parameters
func add(numbers ...int) int {
	sum := 0
	for _, num := range numbers {
		sum += num
	}
	return sum
}

// Main function
func main() {
	// Struct initialization
	test := SyntaxTest{
		name:  "TestUser",
		count: 0,
		items: make([]string, 0, MaxCount),
	}

	// Local variables
	number := 42
	pi := 3.14159
	isActive := true

	// If statement
	if number > 0 {
		fmt.Println(Greeting, test.name)
	} else if number == 0 {
		fmt.Println("Zero detected")
	} else {
		fmt.Println("Negative number")
	}

	// For loop
	for i := 0; i < MaxCount; i++ {
		test.items = append(test.items, fmt.Sprintf("Item %d", i))
	}

	// Switch with type assertion
	var value interface{} = "Go"
	switch v := value.(type) {
	case string:
		fmt.Printf("String value: %s\n", v)
	case int:
		fmt.Printf("Integer value: %d\n", v)
	default:
		fmt.Println("Unknown type")
	}

	// Anonymous function (closure)
	printer := func(s string) {
		fmt.Printf("Closure: %s\n", s)
	}
	for _, item := range test.items {
		printer(item)
	}

	// Error handling
	result, err := divide(10, number)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
	} else {
		fmt.Printf("Result: %d\n", result)
	}

	// Map
	scores := make(map[string]int)
	scores["Alice"] = 95
	scores["Bob"] = 85

	// Goroutine with WaitGroup
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		fmt.Println("Goroutine running")
		time.Sleep(time.Second)
	}()
	wg.Wait()

	// Channel
	ch := make(chan string, 1)
	go func() {
		ch <- "Message from goroutine"
	}()
	fmt.Println(<-ch)

	// Call interface method
	var p Printable = &test
	p.PrintInfo()

	// Defer statement
	defer fmt.Println("Deferred cleanup")
}

// Function with error return
func divide(a, b int) (int, error) {
	if b == 0 {
		return 0, fmt.Errorf("division by zero")
	}
	return a / b, nil
}

// Struct with embedded type
type Config struct {
	SyntaxTest
	enabled bool
}

// Example of init function
func init() {
	fmt.Println("Initializing package")
}

File: snippets/syntax/SyntaxTest.rs
// Sample Rust file for testing VSCode theme syntax highlighting
use std::collections::HashMap;
use std::fmt::Debug;
use std::sync::Arc;

// Constants
const MAX_COUNT: i32 = 100;
const GREETING: &str = "Hello, Rust!";

/// A sample struct to demonstrate syntax highlighting
#[derive(Debug)]
pub struct SyntaxTest<'a> {
    name: &'a str,
    count: i32,
    items: Vec<String>,
}

// Enum
#[derive(Debug)]
enum Color {
    Red,
    Green,
    Blue(u8),
}

// Trait
trait Printable {
    fn print_info(&self);
}

// Struct implementation
impl<'a> SyntaxTest<'a> {
    /// Creates a new SyntaxTest instance
    pub fn new(name: &'a str) -> Self {
        Self {
            name,
            count: 0,
            items: Vec::new(),
        }
    }

    pub fn perform_actions(&mut self) {
        // Local variables
        let number: i32 = 42;
        let pi: f64 = 3.14159;
        let is_active: bool = true;

        // If let
        if let Some(value) = self.increment_count(number) {
            println!("{} {}: {}", GREETING, self.name, value);
        } else {
            println!("No value returned");
        }

        // Match expression
        let color = Color::Blue(255);
        match color {
            Color::Red => println!("Color is red"),
            Color::Green => println!("Color is green"),
            Color::Blue(intensity) => println!("Color is blue with intensity {}", intensity),
        }

        // Loop
        for i in 0..MAX_COUNT {
            self.items.push(format!("Item {}", i));
        }

        // Closure
        let printer = |s: &str| println!("Closure: {}", s);
        self.items.iter().for_each(printer);

        // Result and error handling
        let result = self.divide(10, number);
        match result {
            Ok(value) => println!("Result: {}", value),
            Err(e) => eprintln!("Error: {}", e),
        }

        // Generics and collections
        let mut scores: HashMap<String, i32> = HashMap::new();
        scores.insert(String::from("Alice"), 95);
        scores.insert(String::from("Bob"), 85);
    }

    fn increment_count(&mut self, value: i32) -> Option<i32> {
        self.count += value;
        Some(self.count)
    }

    fn divide(&self, a: i32, b: i32) -> Result<i32, String> {
        if b == 0 {
            Err(String::from("Division by zero"))
        } else {
            Ok(a / b)
        }
    }
}

// Trait implementation
impl<'a> Printable for SyntaxTest<'a> {
    fn print_info(&self) {
        println!("Name: {}, Count: {}, Items: {:?}", self.name, self.count, self.items);
    }
}

// Macro definition
macro_rules! log {
    ($($arg:tt)*) => {
        println!("LOG: {}", format!($($arg)*));
    };
}

// Async function
async fn async_example() -> Result<(), String> {
    log!("Running async example");
    Ok(())
}

// Main function
fn main() {
    let mut test = SyntaxTest::new("TestUser");
    test.perform_actions();
    test.print_info();

    // Using a macro
    log!("Macro test with value: {}", 42);

    // Smart pointer
    let arc = Arc::new(String::from("Shared"));
    let _cloned = Arc::clone(&arc);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_increment_count() {
        let mut test = SyntaxTest::new("Test");
        assert_eq!(test.increment_count(5), Some(5));
    }
}